File: Objects/Row.cs
using System.Collections.Generic;
using System.Drawing;

namespace GanttChart
{
    public class Row
    {
        public Row()
        {
            IsVisible = true;
            TimeBlocks = new List<TimeBlock>();
        }

        public Row(string text)
        {
            this.Text = text;
            IsVisible = true;
            TimeBlocks = new List<TimeBlock>();
        }

        public virtual string Text { get; set; }
        public virtual List<TimeBlock> TimeBlocks { get; set; }
        public virtual Image Icon { get; set; }
        public virtual bool IsVisible { get; set; }
        public virtual Rectangle Rect { get; set; }
        public virtual Rectangle IconRect { get; set; }
    }
}

File: GanttChart/Objects/Row.vb
Imports System.Collections.Generic
Imports System.Drawing

Namespace GanttChart
    Public Class Row
        Public Sub New()
            IsVisible = True
            TimeBlocks = New List(Of TimeBlock)()
        End Sub

        Public Sub New(text As String)
            Me.Text = text
            IsVisible = True
            TimeBlocks = New List(Of TimeBlock)()
        End Sub

        Public Overridable Property Text As String
        Public Overridable Property TimeBlocks As List(Of TimeBlock)
        Public Overridable Property Icon As Image
        Public Overridable Property IsVisible As Boolean
        Public Overridable Property Rect As Rectangle
        Public Overridable Property IconRect As Rectangle
    End Class
End Namespace
